### PR TITLE
fix(gamm): SpotPrice keeper function

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

## What is the purpose of the change

This PR fixes the SpotPrice keeper function.

It fixes the underlying balancer pool bug and stableswap pool hack. Then, it switches parameter order on the keeper layer so that TWAP does not require migrations.

The parameter order is switched for the following functions:
- [`CalculateSpotPrice`](https://github.com/osmosis-labs/osmosis/blob/16eb3e55e3712cffdde9e26c5b22e056b586704a/x/gamm/keeper/pool_service.go#L24)
- [`balancer.SpotPrice`](https://github.com/osmosis-labs/osmosis/blob/16eb3e55e3712cffdde9e26c5b22e056b586704a/x/gamm/pool-models/balancer/pool.go#L625)
- [`stableswap.SpotPrice`](https://github.com/osmosis-labs/osmosis/blob/16eb3e55e3712cffdde9e26c5b22e056b586704a/x/gamm/pool-models/stableswap/pool.go#L334)

It refactors tests that utilize these functions. As an outcome of this change, `v1.SpotPrice` querier [switches ](https://github.com/osmosis-labs/osmosis/blob/09e0ce27ac3e2a70eb185f0a2591cf9fe2d6e3f0/x/gamm/keeper/grpc_query.go#L352-L354)the arguments in the wrong order to make the result correct.

## Testing and Verifying

- Pre-existing twap tests pass
- GAMM tests pass with base / quote switched
- Pool model tests pass with base / quote switched

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? yes
  - How is the feature or change documented? not applicable

### CODE REVIEW

**Review by czarcas7ic — APPROVED** [review]
Changes LGTM

File: x/gamm/pool-models/balancer/pool.go
Comment (nicolaslara) [inline x/gamm/pool-models/balancer/pool.go:622]: This is the same thing, but I find it easier to read "it costs 0.5 uosmo to get one uatom" and "it costs "2 uatom to get 1 osmo".

I think this is because the word "quote" implies to me that it is the price/cost of the base asset denominated in quote asset. So, what you're "getting" is the base asset.

File: x/twap/logic.go
Comment (nicolaslara) [inline x/twap/logic.go:60]: useful to have these comments here!

**Review by nicolaslara — APPROVED** [review]

File: x/gamm/pool-models/balancer/pool.go
Comment (p0mvn) [reply x/gamm/pool-models/balancer/pool.go:623]: Gotcha, I added your interpretation in comments as well

**Comment by p0mvn** [comment]
Merging once CI passes since have 2 approvals

**Pull request merged**
